# Exploration des données

## Importation librairies et dataset

In [5]:
import numpy as np
import pandas as pd
import pickle
import sweetviz as sv

In [6]:
data_lg = pd.read_pickle("datasets/dataset.pkl")
data_sm = pd.read_pickle("datasets/dataset_withoutBadValues.pkl")

In [7]:
data = data_sm
data.info()

<class 'pandas.core.frame.DataFrame'>
Index: 473218 entries, 0 to 899161
Data columns (total 25 columns):
 #   Column         Non-Null Count   Dtype         
---  ------         --------------   -----         
 0   Name           473218 non-null  object        
 1   City           473218 non-null  object        
 2   State          473218 non-null  category      
 3   Zip            473218 non-null  object        
 4   Bank           473218 non-null  object        
 5   BankState      473218 non-null  category      
 6   NAICS          473218 non-null  object        
 7   ApprovalDate   473218 non-null  datetime64[ns]
 8   ApprovalFY     473218 non-null  category      
 9   Term           473218 non-null  int64         
 10  NoEmp          473218 non-null  int64         
 11  NewExist       473218 non-null  category      
 12  CreateJob      473218 non-null  int64         
 13  RetainedJob    473218 non-null  int64         
 14  FranchiseCode  473218 non-null  object        
 15  Urban

In [11]:
for col in data.columns:
    display(col)
    display(data[col].unique())

'Name'

array(['ABC HOBBYCRAFT', 'LANDMARK BAR & GRILLE (THE)',
       'WHITLOCK DDS, TODD M.', ..., 'PULLTARPS MFG.', 'FABRIC FARMS',
       'RADCO MANUFACTURING CO.,INC.'], dtype=object)

'City'

array(['EVANSVILLE', 'NEW PARIS', 'BLOOMINGTON', ..., 'BARATARIA',
       'FRITCH', 'ELLICOTT'], dtype=object)

'State'

['IN', 'CT', 'FL', 'NC', 'IL', ..., 'WV', 'SD', 'DE', 'DC', 'HI']
Length: 51
Categories (51, object): ['AK', 'AL', 'AR', 'AZ', ..., 'WA', 'WI', 'WV', 'WY']

'Zip'

array([47711, 46526, 47401, ..., 66549, 97865, 79036], dtype=object)

'Bank'

array(['FIFTH THIRD BANK', '1ST SOURCE BANK', 'GRANT COUNTY STATE BANK',
       ..., 'MAHOPAC BANK', 'FIRST NATIONAL BANK OF GILBERT',
       'EXCHANGE BANK OF MISSOURI'], dtype=object)

'BankState'

['OH', 'IN', 'DE', 'AL', 'FL', ..., 'AK', 'DC', 'PR', 'GU', 'AN']
Length: 54
Categories (56, object): ['AK', 'AL', 'AN', 'AR', ..., 'WA', 'WI', 'WV', 'WY']

'NAICS'

array([451120, 722410, 621210, ..., 922140, 514110, 221121], dtype=object)

'ApprovalDate'

<DatetimeArray>
['1997-02-28 00:00:00', '2006-02-07 00:00:00', '1997-07-21 00:00:00',
 '1998-05-26 00:00:00', '1998-05-27 00:00:00', '1998-06-01 00:00:00',
 '1998-06-02 00:00:00', '1998-06-05 00:00:00', '1998-06-12 00:00:00',
 '1998-06-15 00:00:00',
 ...
 '1997-02-13 00:00:00', '1997-02-18 00:00:00', '1997-02-14 00:00:00',
 '1997-02-19 00:00:00', '1997-02-20 00:00:00', '1997-02-21 00:00:00',
 '1997-02-24 00:00:00', '1997-02-25 00:00:00', '1997-02-26 00:00:00',
 '1997-02-27 00:00:00']
Length: 7322, dtype: datetime64[ns]

'ApprovalFY'

['1997', '2006', '1998', '1999', '2000', ..., '2013', '1994', '2014', '1995', '1996']
Length: 29
Categories (48, object): ['1966', '1968', '1969', '1970', ..., '2011', '2012', '2013', '2014']

'Term'

array([ 84,  60, 180, 120, 297, 162,  12,  87, 126,  83, 240, 102,  80,
        42,  26,  36,  54,  66,  96,  93,  27,  82,  32, 112,  73, 255,
        55, 133,  35,   5,  62,  72,   6,  57,  46,  70,  52,  48,  65,
        47,  58,   1,  74,  16,   7,   4,  63,  56,  10,  22,   0,  40,
         3,  23, 300,  11,  17,  21,  43,  91,  89,   2,  34,  14,   9,
        78,  18, 129,  97,   8,  29,  30,  39,  24,  13,  15, 119, 228,
        59, 208,  44,  81,  85,  77,  38,  69,  51, 144,  64, 111,  90,
        75, 306,  28, 183, 117, 118,  71,  49, 108, 116,  76, 113, 191,
        19,  61,  45,  99,  25,  98, 166, 114,  94,  95, 231,  33, 157,
       165,  41,  50, 309, 106,  20, 229, 204,  31, 276, 115,  53, 174,
        37,  68,  79, 100, 131, 123,  67, 122, 268, 294, 238, 246, 124,
       186,  92,  86, 192, 128, 147, 280, 318, 264, 101, 121, 216, 278,
       107,  88, 283, 104, 187, 103, 136, 288, 175, 139, 110, 241, 156,
       303, 177, 273, 132, 189, 222, 164, 301, 196, 155, 243, 25

'NoEmp'

array([   4,    2,    7,   19,    1,   24,    5,   18,   12,    3,   20,
          6,    8,   50,   60,   55,   32,   10,   22,  214,   14,    9,
         28,   11,   40,   30,   16,  112,   15,   23,   17,   65,   25,
         45,   97,   46,   13,   31,  126,   58,   38,    0,   75,   36,
         35,   66,   27, 2000,   34,   26,   93,   85,  150,   29,   41,
        100,   54,   21,  300,   49,  120,   33,  265,  160,   70,   68,
         48,   61,  220,   39,   71,   43,   44,   78,  233,  263,   80,
         63,  210,  450,   86,  165, 9992,  424,   51,  190,   52,   47,
         42,   74,  250,   69,   37,   90,  140,  260,  339,   87,   73,
        110,  161,   62,   59,  105,  116,   53,   88,  400,   67,  605,
         57,   95,  130,   81,   82,   64,   94,  144,  155,   98,   56,
         77,  108,  138,  125,  117,  118,  171,  170,  145,  115,  119,
        900,  180,   79,  200,  128,   91,  600,  576,  196,  135,  121,
        101,   96,  350,  129,  141,  142,   72,  3

'NewExist'

['2', '1']
Categories (3, object): ['0', '1', '2']

'CreateJob'

array([   0,    5,    4,    1,    8,   15,    3,   11,    2,   10,    7,
         25,    6,   20,   12,    9,   47,   30,   23,   13,   43,   22,
         18,   21,   14,   82,   80,   70,   19,   33,   27,  150,   38,
         97,   32,   17,   16,   40,   34,   36,   50,   28,   24,   35,
        200,   26,   90,   60,   77,   75,   31,  100,   55,   45,   99,
         48,   29,  120,   69,   95,   42,  160,   37,  600, 1000,   44,
         39,   49,   54,   85,   59,  450,  456,   72,  452,   53,  451,
        198,   79,  454,  136,   63,   98,  455,   76,  152,   41,  225,
        110,  453,  458,  457,  174,   46,   64,  320,  154,  300,   58,
        149,   78,   65,  180,  153,  800,  137,  500,   57,  360,   56,
        122, 1200,   66,  112,  118,   73,  151,  195,   67,  138,  125,
         84,  124, 1711,  131, 1618, 1150,  115,  157,  166,  135,  140,
        210,  252,   61,  116,  102,   74,   71,   62,  129,  105,   52,
        144,   81,  179,  171,  141,  126,   92,   

'RetainedJob'

array([   0,   23,    4,    6,    2,    5,   19,    8,   10,   24,    1,
          3,    7,   15,   12,   11,   17,   14,   20,    9,   28,   38,
         26,   13,   50,   93,   40,   18,   21,   35,  150,   25,   45,
         16,  180,  165,   27,   22,   46,   34,   32,  257,   43,   47,
         80,   70,   62,   33,  400,   37,   30,   31,   55,   60,   36,
         95,   48,   65,  120,   29,   71,  130,   81,   52,   94,   58,
         78,  160,  109,  100,   75,  155,   39,   85,   44,   90,   54,
         64,   41,   61,   69,   66,  210,  107,   97,   51,   42,   83,
         63,  112,   53,   77,   76,   87,   68,  118,  138,   67,  125,
         57,   56,  117,  171,  115,  300,  140,  105,   59,   79,  206,
        128,  250,   72,   49,  135,   92,   86,  404,  200,  110,  320,
        139,   82,  108,   88,  104,  230,  102,   98,  101,   84,  220,
         74,  267,   91,  175,  550,  450,  500,   73,  170,  275,  123,
        195,  116,   99,  610,   96,  187,  235,  1

'FranchiseCode'

array([1, 15100, 10656, ..., 16452, 75985, 42533], dtype=object)

'UrbanRural'

[0, 1, 2]
Categories (3, int64): [0, 1, 2]

'RevLineCr'

['N', 'Y']
Categories (3, object): ['0', 'N', 'Y']

'LowDoc'

['Y', 'N']
Categories (3, object): ['0', 'N', 'Y']

'MIS_Status'

array([1, 0])

'GrAppv'

array([ 60000,  40000, 287000, ...,  20686,  45720, 279145])

'SBA_Appv'

array([ 48000,  32000, 215250, ...,  36576, 209358, 430976])

'Zip2'

['47', '46', '60', '34', '32', ..., '7', '8', '5', '3', '6']
Length: 99
Categories (100, object): ['0', '1', '10', '11', ..., '96', '97', '98', '99']

'NAICS2'

['45', '72', '62', '33', '81', ..., '92', '22', '49', '55', '21']
Length: 24
Categories (25, object): ['0', '11', '21', '22', ..., '71', '72', '81', '92']

'Franchised'

[0, 1]
Categories (2, int64): [0, 1]

'RealEstate'

[0, 1]
Categories (2, int64): [0, 1]

## Description des features

| Colonne        | Type de données   | Modalités                 | Exemple de Données | Description                                       |
|----------------|-------------------|---------------------------|--------------------|---------------------------------------------------|
| Name           | Object            | N/A                       | "Entreprise A"     | Nom de l'entreprise souhaitant réaliser le prêt   |
| City           | Object            | N/A                       | "New York"         | Ville où se l'entreprise                          |
| State          | Category          | "NY", "CA" ...            | "NY"               | État où se situe l'entreprise (abréviations en deux lettres)|
| Zip            | Object            | N/A                       | 10001              | Code postal de l'entreprise                       |
| Bank           | Object            | N/A                       | "ABC Bank"         | Nom de la banque accordant le prêt                |
| BankState      | Category          | "NY", "CA"...             | "NY"               | État où se situe la banque (abréviations en deux lettres)|
| NAICS          | Object            | N/A                       | 722410             | Code du secteur d'activité de l'entreprise        |
| ApprovalDate   | Datetime64[ns]    | N/A                       | 2023-05-12         | Date à laquelle le prêt a été approuvé            |
| ApprovalFY     | Category          | "1970", "1971" ..         | "2023"             | Année fiscale du prêt                             |
| Term           | Int64             | N/A                       | 60                 | Durée du prêt en mois                             |
| NoEmp          | Int64             | N/A                       | 10                 | Nombre d'employés de l'entreprise                 |
| NewExist       | Category          | "New", "Existing"         | "Existing"         | Indication si l'entreprise est nouvelle ou existante|
| CreateJob      | Int64             | N/A                       | 5                  | Nombre d'emplois créés par le prêt                |
| RetainedJob    | Int64             | N/A                       | 8                  | Nombre d'emplois conservés grâce au prêt          |
| FranchiseCode  | Object            | 0, 11500, 12564 ...       | 0                  | Code indiquant si l'entreprise est une franchise  |
| UrbanRural     | Category          | "Urban", "Rural",         | Urban              | Type d'environnement où se situe l'entreprise     |
| RevLineCr      | Category          | "Y", "N"                  | Y                  | Indication sur si une ligne de crédit rotatif est disponible |
| LowDoc         | Category          | "Y", "N"                  | N                  | Indication sur si le prêt est de faible documentation |
| MIS_Status     | Int64             | N/A                       | 1                  | Statut du prêt                                    |
| GrAppv         | Int64             | N/A                       | 150000             | Montant total approuvé pour le prêt               |
| SBA_Appv       | Int64             | N/A                       | 75000              | Montant approuvé par la Small Business Administration pour le prêt |
| Zip2           | Category          | "10", "20", "30", ...     | 10                 | Code postal de l'entité (représenté en catégorie) |
| NAICS2         | Category          | "120000", "130000"        | 120000             | Code du secteur d'activité de l'entité (représenté en catégorie) |
| Franchised     | Category          | "Yes", "No"               | No                 | Indication sur si l'entité est une franchise      |
| RealEstate     | Category          | "Yes", "No"               | Yes                | Indication sur si le prêt est lié à l'immobilier  |



## Exploration avec SweetViz

In [8]:
report = sv.analyze(data, "MIS_Status")
report.show_html("sv-reports/report_sm.html", layout="vertical")

                                             |          | [  0%]   00:00 -> (? left)

Done! Use 'show' commands to display/save.   |██████████| [100%]   00:01 -> (00:00 left)


Report sv-reports/report_sm.html was generated! NOTEBOOK/COLAB USERS: the web browser MAY not pop up, regardless, the report IS saved in your notebook/colab files.


## Autres explorations

In [5]:
subway_subset = data[data["Name"] == "SUBWAY"]

display(subway_subset.head())

subway_subset["MIS_Status"].value_counts()


Name         City State    Zip                            Bank  \
24     SUBWAY  LITTLE ROCK    AR  72223                        HOPE FCU   
3156   SUBWAY   GLEN MILLS    PA  19342  PNC BANK, NATIONAL ASSOCIATION   
7461   SUBWAY   MANCHESTER    ME   4351             BANGOR SAVINGS BANK   
8287   SUBWAY      NORWICH    CT   6360      BANK OF AMERICA NATL ASSOC   
10198  SUBWAY       ATHENS    TX  75751  COMMUNITY NATL BK & TR OF TEXA   

      BankState   NAICS ApprovalDate ApprovalFY  Term  ...  UrbanRural  \
24           MS  722211   2006-02-07       2006   126  ...           1   
3156         DE  722211   2006-02-10       2006    72  ...           1   
7461         ME  722211   2006-02-16       2006    84  ...           1   
8287         RI  722211   2004-09-08       2004    36  ...           1   
10198        TX       0   1997-04-01       1997    84  ...           0   

      RevLineCr  LowDoc  MIS_Status  GrAppv SBA_Appv Zip_2 NAICS_2  \
24            N       N           1  137300   116705    72      72   
3156          0       N           1   50000    25000    19      72   
7461          0       N           1  102000    86700    43      72   
8287          Y       N           1   47000    23500    63      72   
10198         N       Y           1   77600    62080    75       0   

       Franchised  RealEstate  
24              0           0  
3156            0           0  
7461            1           0  
8287            1           0  
10198           0           0  

[5 rows x 25 columns]

MIS_Status
1    1203
0      66
Name: count, dtype: int64

In [6]:
pizza_subset = data[data["Name"] == "DOMINO'S PIZZA"]

display(pizza_subset.head())

Name               City State    Zip  \
2499   DOMINO'S PIZZA           SARASOTA    FL  34231   
10553  DOMINO'S PIZZA        SPRINGFIELD    MO  65802   
11440  DOMINO'S PIZZA  PONTE VERDE BEACH    FL  32082   
12269  DOMINO'S PIZZA       GUNTERSVILLE    AL  35976   
14721  DOMINO'S PIZZA          UNIONDALE    NY  11553   

                              Bank BankState NAICS ApprovalDate ApprovalFY  \
2499         STEARNS BK NATL ASSOC        FL     0   1997-03-06       1997   
10553                  EMPIRE BANK        MO     0   1997-03-28       1997   
11440             OLD FLORIDA BANK        FL     0   1997-04-01       1997   
12269            BANCORPSOUTH BANK        AL     0   1997-04-02       1997   
14721  BANCO POPULAR NORTH AMERICA        NY     0   1997-04-08       1997   

       Term  ...  UrbanRural RevLineCr  LowDoc  MIS_Status  GrAppv SBA_Appv  \
2499     60  ...           0         N       N           1  260000   195000   
10553    84  ...           0         N       N           1   90000    72000   
11440    85  ...           0         0       N           1   90000    72000   
12269    60  ...           0         N       Y           1   75000    60000   
14721   120  ...           0         0       N           1   96000    76800   

      Zip_2 NAICS_2  Franchised  RealEstate  
2499     34       0           1           0  
10553    65       0           1           0  
11440    32       0           1           0  
12269    35       0           1           0  
14721    11       0           1           0  

[5 rows x 25 columns]

In [7]:
naics_subset = data[data["NAICS"] == 0]
display(naics_subset)

Name          City State    Zip  \
3        BIG BUCKS PAWN & JEWELRY, LLC  BROKEN ARROW    OK  74012   
4          ANASTASIA CONFECTIONS, INC.       ORLANDO    FL  32801   
6        MIDDLE ATLANTIC SPORTS CO INC         UNION    NJ   7083   
9              INTEXT BUILDING SYS LLC   GLASTONBURY    CT   6073   
13         ORCHARD CAFE & BAKERY, INC.  SLATERSVILLE    RI   2876   
...                                ...           ...   ...    ...   
899153              NORTH SHORE FLORAL      WOODBURY    NY  11797   
899154    LITWIN LIVERY SERVICES, INC.      CAMPBELL    OH  44405   
899158  SHADES WINDOW TINTING AUTO ALA        IRVING    TX  75062   
899162           MARUTAMA HAWAII, INC.      HONOLULU    HI  96830   
899163  PACIFIC TRADEWINDS FAN & LIGHT        KAILUA    HI  96734   

                                  Bank BankState NAICS ApprovalDate  \
3        1ST NATL BK & TR CO OF BROKEN        OK     0   1997-02-28   
4              FLORIDA BUS. DEVEL CORP        FL     0   1997-02-28   
6          WELLS FARGO BANK NATL ASSOC        SD     0   1980-06-02   
9              WEBSTER BANK NATL ASSOC        CT     0   1997-02-28   
13            CITIZENS BANK NATL ASSOC        RI     0   1997-02-28   
...                                ...       ...   ...          ...   
899153                   FLUSHING BANK        NY     0   1997-02-27   
899154  JPMORGAN CHASE BANK NATL ASSOC        IL     0   1997-02-27   
899158   LOANS FROM OLD CLOSED LENDERS        DC     0   1997-02-27   
899162                  BANK OF HAWAII        HI     0   1997-02-27   
899163            CENTRAL PACIFIC BANK        HI     0   1997-02-27   

       ApprovalFY  Term  ...  UrbanRural RevLineCr  LowDoc  MIS_Status  \
3            1997    60  ...           0         N       Y           1   
4            1997   240  ...           0         N       N           1   
6            1980    45  ...           0         N       N           0   
9            1997    84  ...           0         N       Y           1   
13           1997   120  ...           0         N       N           1   
...           ...   ...  ...         ...       ...     ...         ...   
899153       1997   119  ...           0         0       N           0   
899154       1997    60  ...           0         0       N           1   
899158       1997    84  ...           0         N       Y           1   
899162       1997    60  ...           0         N       Y           0   
899163       1997    48  ...           0         N       N           1   

        GrAppv SBA_Appv Zip_2 NAICS_2  Franchised  RealEstate  
3        35000    28000    74       0           0           0  
4       229000   229000    32       0           0           1  
6       600000   499998    70       0           0           0  
9        70000    56000    60       0           0           0  
13      370000   277500    28       0           0           0  
...        ...      ...   ...     ...         ...         ...  
899153  142000   106500    11       0           0           0  
899154   10000     5000    44       0           0           0  
899158   79000    63200    75       0           0           0  
899162   75000    60000    96       0           0           0  
899163   30000    24000    96       0           0           0  

[201667 rows x 25 columns]